### PR TITLE
Add PNG support

### PR DESCRIPTION
**Pull request by Ralender**
I don't know how we should integrate switch between ppm and png.

but I think png is a better default than ppm.

### CODE REVIEW

**Comment by lforg37** [comment]
> but I think png is a better default than ppm.

I'm also in favor of using png.

> I don't know how we should integrate switch between ppm and png.

I think it is reasonable to just completely drop the ppm support given the state of the project.

If we switch to png the filename extension in the documentation example should also be modified

File: src/main.cpp [hunk 1]
```diff
@@ -8,10 +8,13 @@
...
 #include <vector>
 
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb/stb_image_write.h>
+
 #include "render.hpp"
 
 // Function to save image data in ppm format
-void save_image(int width, int height, auto& fb_data) {
+void dump_image_ppm(int width, int height, auto& fb_data) {
```
Comment (lforg37) [inline src/main.cpp:17]: ``dump_image_ppm` vs `save_image_png`.

But I suggest we keep only the png version under the name `save_image`

**Review by lforg37 — COMMENTED** [review]
`dump_image_ppm` vs `save_image_png` -> either we agree to keep only the png version, or we should harmonize the names.

Filename extension in the doc should be modified if we stay on png.

File: src/main.cpp [hunk 2]
```diff
@@ -27,6 +30,33 @@ void save_image(int width, int height, auto& fb_data) {
...
+      int r = static_cast<int>(
+          256 * std::clamp(sycl::sqrt(fb_data[input_index].x()), 0.0f, 0.999f));
+      int g = static_cast<int>(
+          256 * std::clamp(sycl::sqrt(fb_data[input_index].y()), 0.0f, 0.999f));
+      int b = static_cast<int>(
+          256 * std::clamp(sycl::sqrt(fb_data[input_index].z()), 0.0f, 0.999f));
+
+      pixels[index++] = r;
+      pixels[index++] = g;
+      pixels[index++] = b;
```
Comment (keryell) [inline src/main.cpp:52]: All this processing should be done in the kernel in some future versions.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**